#!/bin/bash
# Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.

--- ./utils/aperf.c.old	2015-03-09 14:21:22.474696191 -0500
+++ ./utils/aperf.c	2015-03-09 14:22:01.394697321 -0500
@@ -397,10 +397,13 @@
 		}
 	}
 
-	if (getuid() != 0) {
-		fprintf(stderr, "You must be root\n");
+	if (geteuid() != 0) {
+		fprintf(stderr, "You must be root %d\n", geteuid());
 		return EXIT_FAILURE;
 	}
+	if (getuid() != 0) {
+		setuid(0);
+	}
 
 	if (!cpu_has_effective_freq()) {
 		fprintf(stderr, "CPU doesn't support APERF/MPERF\n");
